### ISSUE TITLE
[type-annotation-spacing] Number of spaces aren't enforced

### ISSUE THREAD

**Issue by rhalaly** [typescript-eslint/typescript-eslint#1001]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/type-annotation-spacing": "error"
  }
}
```

```TS
class Foo {
    private bar:          Bar;
}
```

**Expected Result**
Error for member filed `bar`, that have more than one space after colon.

**Actual Result**
No error

**Additional Info**
The current rule enforces the spaces as described, but it doesn't enforce how much spaces there is.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.1.0` |
| `@typescript-eslint/parser`               | `2.1.0` |
| `TypeScript`                              | `3.5.3` |
| `ESLint`                                  | `6.3.0` |
| `node`                                    | `8.11.3` |
| `npm`                                     | `6.11.2` |

**Comment by vinaysharma14** [comment]
You can use `"no-multi-spaces": "error",`

**Comment by rhalaly** [comment]
I thought this rule should enforce multiline as well... But thank you

**Comment by rhalaly** [comment]
Sorry, typo :). Meant multi spaces enforcement...

**Comment by armano2** [comment]
this rule actually works as described in documentation and similarity to other spacing rules from eslint
spacing rules except `no-multi-spaces` enforce no whitespace or at least one whitespace between tokens

https://eslint.org/docs/rules/template-curly-spacing
https://eslint.org/docs/rules/generator-star-spacing
https://eslint.org/docs/rules/array-bracket-spacing

**Comment by lonyele** [comment]
hm... should this rule also error when there is more than 1 space? If that's the case, I could make a PR

**Comment by Josh-Cena** [comment]
That would be my thought, yeah

**Comment by JoshuaKGoldberg** [comment]
Oof, yes, sorry for marking this as `accepting prs` - that was incorrect of me. Thanks for the catch.
